src/compiler/checker.ts

```diff
@@ -12495,6 +12509,36 @@ namespace ts {
         }
 
         /**
+         * Checks that the return type provided is an instantiation of the global Promise<T> type

```

Comment:
What is the reason for the compiler to reject type annotations that are known compatible with `Promise`? This won't catch any type errors, but rejects valid annotations. Generator functions handle this consistently IMO. Why not use the same approach here for consistency?

```typescript
function* bar() {}                  // Returns IterableIterator<any>
function* bar2(): any {}            // OK: compatible annotation
function* bar3(): Iterator<any> {}  // OK: comtatible annotation
function* bar4(): { next } {}       // OK: comtatible annotation

async function foo() {}                     // Returns Promise<void>
async function foo1(): any {}               // ERROR with this PR, but why?
async function foo2(): PromiseLike<any> {}  // ERROR with this PR. but why?
```